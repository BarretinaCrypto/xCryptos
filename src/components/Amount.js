import { array, number, oneOfType, shape } from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { View, Text } from 'react-native';
import { C, SHAPE, STYLE } from '../config';
import { formatCurrency } from '../modules';

const { SETTINGS } = SHAPE;
const { CURRENCY: { USD }, SYMBOL } = C;

const symbolStyle = {
  opacity: 0.75,
  transform: [{ scale: 0.75 }],
};

export const Amount = ({ settings: { currency }, style, value }) => (
  <View style={STYLE.ROW}>
    { currency === USD && <Text style={[style, symbolStyle]}>{SYMBOL.USD}</Text> }
    <Text style={style}>{formatCurrency(value)}</Text>
    { currency !== USD && <Text style={[style, symbolStyle]}>{SYMBOL[currency]}</Text> }
  </View>
);

Amount.propTypes = {
  settings: shape(SETTINGS),
  style: oneOfType([array, number]),
  value: number,
};

Amount.defaultProps = {
  settings: {},
  style: [],
  value: 0,
};

const mapStateToProps = ({ settings }) => ({
  settings,
});

export default connect(mapStateToProps)(Amount);
