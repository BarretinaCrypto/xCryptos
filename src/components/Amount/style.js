import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  symbol: {
    opacity: 0.75,
    transform: [{ scale: 0.75 }],
  },
});
