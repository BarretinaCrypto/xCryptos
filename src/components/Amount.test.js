import 'react-native';
import React from 'react';
import { Amount } from './Amount';
import renderer from 'react-test-renderer';

it('renders by default', () => {
  const tree = renderer.create(<Amount />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with a custom amount', () => {
  const tree = renderer.create(<Amount value={7.77} />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('when currency is USD', () => {
  it('renders currency before amount', () => {
    const currency = 'USD';
    const tree = renderer.create(<Amount settings={{ currency }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('when currency is not USD', () => {
  it('renders currency after amount', () => {   
    const currency = 'EUR';
    const tree = renderer.create(<Amount settings={{ currency }} />).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});
