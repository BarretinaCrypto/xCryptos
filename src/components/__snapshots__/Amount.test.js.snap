// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders by default 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Array []}
  >
    0
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Array [],
        Object {
          "opacity": 0.75,
          "transform": Array [
            Object {
              "scale": 0.75,
            },
          ],
        },
      ]
    }
  />
</View>
`;

exports[`renders with a custom amount 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Array []}
  >
    7.77
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Array [],
        Object {
          "opacity": 0.75,
          "transform": Array [
            Object {
              "scale": 0.75,
            },
          ],
        },
      ]
    }
  />
</View>
`;

exports[`when currency is USD renders currency before amount 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Array [],
        Object {
          "opacity": 0.75,
          "transform": Array [
            Object {
              "scale": 0.75,
            },
          ],
        },
      ]
    }
  >
    $
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Array []}
  >
    0
  </Text>
</View>
`;

exports[`when currency is not USD renders currency after amount 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Array []}
  >
    0
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Array [],
        Object {
          "opacity": 0.75,
          "transform": Array [
            Object {
              "scale": 0.75,
            },
          ],
        },
      ]
    }
  >
    â‚¬
  </Text>
</View>
`;
